core/trino-main/src/main/java/io/trino/sql/planner/DistributedExecutionPlanner.java

```diff
@@ -203,12 +207,21 @@ private Visitor(
                 dynamicFilter = dynamicFilterService.createDynamicFilter(session.getQueryId(), dynamicFilters, node.getAssignments(), typeProvider);
             }
 
+            ConstraintEvaluator constraintEvaluator = ConstraintEvaluator.EMPTY;
+            Optional<ConstraintEvaluator> existingEvaluator = filter.map(FilterNode::getPredicate)
+                    .filter(expression -> !DynamicFilters.isDynamicFilter(expression))
```

Comment:
You need to disassemble the `expression` into conjuncts, filtering out dynamic filters

```
Expression nonDynamicPredicate = [ExpressionUtils.]filterConjuncts(metadata, predicate, expression -> !DynamicFilters.isDynamicFilter(expression));
```